locust_plugins/transaction_manager.py

```diff
@@ -0,0 +1,238 @@
+from locust import events
+from time import time
+from datetime import datetime
+import locust.stats
+
+
+class TransactionManager:
+    """
+    Transaction Manager allows transactions spanning multiple tasks to be logged
+    using start_transaction and end_transaction methods
+    Stats are written to file when using --log_transactions_in_file=True
+    otherwise, two web endpoints are available to collect full and summary stats:
+    /stats/transactions/csv and /stats/transactions/all/csv
+    when running in master/worker mode, all data is sent to the master during execution
+    """
+
+    flush_size = 100
+    field_delimiter = ","
+    row_delimiter = "\n"
+    timestamp_format = "%Y-%m-%d %H:%M:%S"
+    flat_transaction_list = []
+    completed_transactions = {}
+    test_timestamp = datetime.fromtimestamp(time()).strftime("%Y_%m_%d_%H_%M_%S")
+    transactions_filename = f"transactions_{test_timestamp}.csv"
+    transactions_summary_filename = f"transactions_summary_{test_timestamp}.csv"
+    user_count = 0
+    csv_headers = [
+        "start_time",
+        "duration",
+        "transaction_name",
+        "user_count",
+        "success",
+        "failure_message",
+    ]
+
+    def __init__(self):
+        self.transaction_count = 0
+        self.inprogress_transactions = {}
+        # results filename format
+        # fields set for file output
+
+    @classmethod
+    def _command_line_parser(cls, parser):
+        parser.add_argument(
+            "--log_transactions_in_file",
+            help="To log transactions in file rather than using the web ui, set to True",
+            default=False,
+        )
+        cls.is_not_worker = not parser.parse_args().worker
```

Comment:
You should be able to check whether you are in a worker by checking the type of environment.runner instead.

Reply:
thanks - I'll change it for consistency with the other plugins